src/components/data_comps/datm/datm_comp_mod.F90

```diff
@@ -722,13 +720,14 @@ subroutine datm_comp_run(EClock, x2a, a2x, &
     enddo
     call t_stopf('datm_scatter')
 
-    call t_startf('datm_mode')
-    select case (trim(atm_mode))
+    !-------------------------------------------------
+    ! Determine data model behavior based on the mode
+    !-------------------------------------------------
 
-    case('COPYALL')
-       ! do nothing extra
+    call t_startf('datm_datamode')
+    select case (trim(datamode))
 
-    case('CPLHIST')
+    case('COPYALL')
```

Comment:
Is there going to be confusion or challenges upgrading if CPLHIST is changed to COPYALL?

Reply:
No - this is the change that I made in the esmci master as well.  Since it behaves as copyall - there is no reason to introduce a different datamode unless we have to. This minimizes confusion.

Reply:
OK, I see, you are just passing in the array, not the attribute vector.  The attribute vector has the field names.  Any reason not to keep the AvectToState version around?

Reply:
No - that is now gone. There is no longer a shr_nuopc_dmodel_mod (see item 2. in my summary)